addons/stock/models/stock_warehouse.py

```diff
@@ -1025,6 +1025,11 @@ def default_get(self, fields):
         ('qty_multiple_check', 'CHECK( qty_multiple >= 0 )', 'Qty Multiple must be greater than or equal to zero.'),
     ]
 
+    @api.model
+    def create(self, vals):
+        vals['name'] = self.env['ir.sequence'].next_by_code('stock.orderpoint')
```

Comment:
Check if vals['name'] == _('New')
Otherwise you will prevent user to use a name that doesn't come from the ir.sequence

Reply:
done, thanks!